### PR TITLE
feat: Build linux static executable

### PR DESCRIPTION
**Pull request by shivaraj-bh**
Verify if the executable is statically linked:
```sh
ldd $(nix build github:shivaraj-bh/nix-health/static-linux --print-out-paths)/bin/nix-health
```

TODO
- [x] Use build flags and target ENVs only if `pkgs.stdenv.isLinux`
- [x] Fix checks in CI (https://github.com/juspay/nix-health/actions/runs/9563018190/job/26360637857?pr=23)

### CODE REVIEW

**Comment by shivaraj-bh** [comment]
Checks seem to fail due to docs. This was fixed in https://github.com/ipetkov/crane/pull/603

**Comment by shivaraj-bh** [comment]
https://github.com/juspay/rust-flake/pull/8 should fix the cargo docs

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,2 +1,3 @@
 [toolchain]
 channel = "nightly"
+targets = [ "x86_64-unknown-linux-musl" ]
```
Comment (srid) [inline rust-toolchain.toml:3]: This won't negatively impact development on Linux (e.g.: `nix develop -c cargo run`), right?

**Review by srid — COMMENTED** [review]
LGTM, once the flake inputs are updated.

File: rust-toolchain.toml (hunk 1 above)
Comment (shivaraj-bh) [reply rust-toolchain.toml:3]: Seems to be working fine for me

**Comment by shivaraj-bh** [comment]
> LGTM, once the flake inputs are updated.

Done

**Pull request merged**